lib/jsdom/living/events/EventTarget-impl.js

```diff
@@ -322,13 +315,7 @@ function innerInvokeEventListeners(eventImpl, listeners, phase) {
     }
 
     try {
-      if (typeof listener.callback === "object") {
-        if (typeof listener.callback.handleEvent === "function") {
-          listener.callback.handleEvent(idlUtils.wrapperForImpl(eventImpl));
-        }
-      } else {
-        listener.callback.call(eventImpl.currentTarget, idlUtils.wrapperForImpl(eventImpl));
-      }
+      listener.callback.call(eventImpl.currentTarget, idlUtils.wrapperForImpl(eventImpl));
```

Comment:
The `wrapperForImpl` would no longer be necessary, right?

Reply:
Apparently, that caused some weird issues with the way things interact in **JSDOM**: <https://travis-ci.com/github/jsdom/jsdom/builds/158224937>.

Probably something somewhere registers a listener that expects an event wrapper directly on the implementation, rather than going through the wrapper.

Reply:
Looks like it's https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/helpers/create-event-accessor.js#L11. `el` is an impl, so `event` there is already an impl.

Reply:
It’s something else because I fixed that in <https://github.com/jsdom/jsdom/pull/2884/commits/6bfe5ab875819df23e055523cd026290196fb596#diff-32bfa972748b97cd78336775200b2e64>, but had to roll it back in <https://github.com/jsdom/jsdom/pull/2884/commits/3a2ca8285078c602e14e509006512be405f692ee> due to test failures.

Reply:
Could you then add a `if (!event) throw new Error();` to after https://github.com/jsdom/jsdom/pull/2884/commits/3a2ca8285078c602e14e509006512be405f692ee#diff-32bfa972748b97cd78336775200b2e64R11, so that you can catch that?

Reply:
It’d be so much easier if we had the codebase annotated with **JSDoc**/**TypeScript** types, because then static analysis would make it possible to find the root of the issue much quicker.